homeassistant/components/reolink/camera.py

```diff
@@ -0,0 +1,221 @@
+"""This component provides support for Reolink IP cameras."""
+from __future__ import annotations
+
+from datetime import datetime
+import logging
+from typing import Any, cast
+
+import voluptuous as vol
+
+from homeassistant.components.camera import Camera, CameraEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv, entity_platform
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    DOMAIN,
+    HOST,
+    SERVICE_PTZ_CONTROL,
+    SERVICE_SET_BACKLIGHT,
+    SERVICE_SET_DAYNIGHT,
+    SERVICE_SET_SENSITIVITY,
+    SUPPORT_PTZ,
+)
+from .entity import ReolinkCoordinatorEntity
+from .host import ReolinkHost
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Set up a Reolink IP Camera."""
+    platform = entity_platform.async_get_current_platform()
+    platform.async_register_entity_service(
+        SERVICE_SET_SENSITIVITY,
+        {
+            vol.Required("sensitivity"): cv.positive_int,
+            vol.Optional("preset"): cv.positive_int,
+        },
+        SERVICE_SET_SENSITIVITY,
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_SET_DAYNIGHT,
+        {vol.Required("mode"): cv.string},
+        SERVICE_SET_DAYNIGHT,
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_SET_BACKLIGHT,
+        {vol.Required("mode"): cv.string},
+        SERVICE_SET_BACKLIGHT,
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_PTZ_CONTROL,
+        {
+            vol.Required("command"): cv.string,
+            vol.Optional("preset"): cv.positive_int,
+            vol.Optional("speed"): cv.positive_int,
+        },
+        SERVICE_PTZ_CONTROL,
+        [SUPPORT_PTZ],
+    )
+
```

Comment:
Lets leave out the custom services in the first PR to keep this smaller

Reply:
I have removed everything related to the custom services.